### PR TITLE
Project Overhaul

### PR DESCRIPTION
**Pull request by tolauwae**

### CODE REVIEW

**Comment by beardhatcode** [comment]
I think it might be interesting to move the [vs-debugger](https://github.com/TOPLLab/WARDuino/tree/cleanup/vs-debugger) into a separate repository like TOPLLab/WARDuino-VSC or something like that. What do you think @tolauwae  and @chscholl ?

**Comment by tolauwae** [comment]
> I think it might be interesting to move the [vs-debugger](https://github.com/TOPLLab/WARDuino/tree/cleanup/vs-debugger) into a separate repository like TOPLLab/WARDuino-VSC or something like that. What do you think @tolauwae and @chscholl ?

I too think that logically these should be separate. It also makes more sense to link to the plugin repo than the whole project VM when we publish the VS Code plugin in, hopefully, the near future. 

But the practical reason they are not, is so the CI can immediately alert us when a change to the VM  breaks something with the plugin. So unless we can do this with two seperate repositories as well, I would keep everything in the same repository.

**Comment by tolauwae** [comment]
Moved plugin to new repo: https://github.com/TOPLLab/WARDuino-VSCode

File: examples/assemblyscript/lib/warduino.ts [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+    _print(String.UTF8.encode(text, true), String.UTF8.byteLength(text, true));
+}
+
+export function wifi_connect(ssid: string, password: string): void {
+    _wifi_connect(String.UTF8.encode(ssid, true), String.UTF8.byteLength(ssid, true),
+                  String.UTF8.encode(password, true), String.UTF8.byteLength(password, true));
+}
+
+export function wifi_localip(): string {
+    let localip = new ArrayBuffer(20);
```
Comment (beardhatcode) [inline examples/assemblyscript/lib/warduino.ts:62]:
```suggestion
    let localip = new ArrayBuffer(46);
```

Maybe this should be set tot 46 to also support ipv6 adresses (https://stackoverflow.com/a/166157)

File: src/WARDuino/WARDuino.cpp [hunk 2]
```diff
@@ -823,37 +823,51 @@ Module *WARDuino::load_module(uint8_t *bytes, uint32_t byte_count,
     }
 
     find_blocks(m);
+    debug("findblocks finished\n");
 
     if (m->start_function != UNDEF) {
+        debug("has startfun \n");
         uint32_t fidx = m->start_function;
+        debug("1 \n");
```
Comment (beardhatcode) [inline src/WARDuino/WARDuino.cpp:831]: What are these messages for?

Comment (tolauwae) [reply src/WARDuino/WARDuino.cpp:831]: These were added in commit 23ae473ef1e6f7ac45998d7d0650e525406fff36. I suspect these were never meant to be commited. I will remove them.

File: README.md [hunk 3]
```diff
@@ -1,134 +1,117 @@
...
+This will build the command-line tool (EMULATOR).
 
-```shell
-BOARD=ESP32WROVER
-../../scripts/upload $BOARD hello_world.ino
-```
+### Build for ESP-IDF
 
-  - Use board `DOIT ESP32 DEVKIT V1`
+Before you can compile and flash with ESP-IDF, you must install and enable the toolchain.
```
[the remark this answers is not in the document]
Comment (tolauwae) [reply README.md]: Added in 1fe5979

File: examples/assemblyscript/lib/warduino.ts (hunk 1 above)
Comment (tolauwae) [reply examples/assemblyscript/lib/warduino.ts]: Added in 1fe5979

File: src/WARDuino/WARDuino.cpp (hunk 2 above)
Comment (tolauwae) [reply src/WARDuino/WARDuino.cpp]: Removed in 1fe5979

File: README.md [hunk 4]
```diff
@@ -1,134 +1,117 @@
...
+The WARDuino virtual machine is a WebAssembly runtime for microcontrollers, which runs both under the Arduino and ESP-IDF toolchains.
+The WARDuino project also includes a VS Code extension to use the remote debugging facilities offered by the virtual machine.
 
-## ToolChain Installation
+<p align="center">
+  <a href="./README.md#build-and-development-instructions">Installation</a> | <a href="./examples/">Examples</a> | <a href="./README.md#webassembly-specification-tests">Run Specification tests</a> | <a href="./documentation/">Documentation</a>
+</p>
 
-Follow these steps to execute the blink program on WARDuino on an ESP8266 or ESP32.
+## Build and Development Instructions
```
[the remark this answers is not in the document]
Comment (tolauwae) [reply README.md:18]: Added in 1fe5979

**Review by beardhatcode — APPROVED** [review]
LGTM, nice work!

**Pull request merged**